### PR TITLE
[CAS-428] Message options2

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-428](https://stream-io.atlassian.net/browse/CAS-428)

### Description

First iteration of messages options. This PR creates this view:

Dec-01-2020 23-33-00

The other view at yet to be created. They need to be added to container for the views. 

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes. **Not needed.** 
- [ ] New code is covered by unit tests. **Not needed.** 
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/view/MessageListView.kt [hunk 1]
```diff
@@ -160,6 +162,19 @@ public class MessageListView : ConstraintLayout {
                 onMessageFlagHandler
             ).show()
         }
+    //Todo: Add this after the behaviour of message options is complete.
```
Comment (leandroBorgesFerreira) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/view/MessageListView.kt:165]: If I put this... then I would change the original behaviour before the new one is completed... Not sure if the comment is the best solution, thought...

**Comment by adasiewiczr** [comment]
Shouldn't this be a part of the new `MessageList` component?

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ internal class MessageTextConfigurator(
     private val style: MessageListViewStyle,
     private val bubbleHelper: MessageListView.BubbleHelper,
     private val messageClickListener: MessageListView.MessageClickListener,
-    private val messageLongClickListener: MessageListView.MessageLongClickListener,
+    private var messageLongClickListener: MessageListView.MessageLongClickListener,
```
Comment (JcMinarro) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt:37]: Do we need it as mutable field?
It seems that is not been updated

**Comment by leandroBorgesFerreira** [comment]
@adasiewiczr I thought this was yet to be done. I tried to find a new one but I could not, I don't see a class called:`MessageList`. Which class is the new MessageListView?

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt (hunk 2 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt:37]: No need. This one passed me by.

**Comment by leandroBorgesFerreira** [comment]
Moving this back to Draft. I'll copy the `MessageListView` to the new module so I can change its behaviour.

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/view/MessageListView.kt (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/view/MessageListView.kt]: Removrd

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/view/EndlessScrollListener.kt [hunk 3]
```diff
@@ -3,12 +3,12 @@ package com.getstream.sdk.chat.view
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 
-internal class EndlessScrollListener(
```
Comment (leandroBorgesFerreira) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/view/EndlessScrollListener.kt:6]: I had to change the visibility of some classes and accessors in many classes in other to clone the MessageListView.  It is big change, but I don't see a way to clone MessageListView without doing that or make a super huge PR that brings a lot of classes to the new module

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/ListenerContainerImpl.kt [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
+import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListenerView
+import com.getstream.sdk.chat.view.MessageListView.MessageRetryListener
+import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
+import com.getstream.sdk.chat.view.MessageListView.UserClickListener
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+public class ListenerContainerImpl(
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/ListenerContainerImpl.kt:16]: Just a copy and paste from the ListenerContainerImpl in our old UI components module

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt [hunk 5]
```diff
@@ -0,0 +1,828 @@
...
+
+/**
+ * MessageListView renders a list of messages and extends the [RecyclerView]
+ * The most common customizations are
+ * - Disabling Reactions
+ * - Disabling Threads
+ * - Customizing the click and longCLick (via the adapter)
+ * - The list_item_message template to use (perhaps, multiple ones...?)
+ */
+public class MessageListView : ConstraintLayout, IMessageListView {
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt:64]: Mostly a copy and past from the old MessageListView, with the change in the long click listener

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**